src/univariate/continuous/normal.jl

```diff
@@ -165,30 +167,30 @@ invlogcdf(d::Normal, lp::Real) = xval(d, norminvlogcdf(lp))
 # invlogccdf
 invlogccdf(d::Normal, lp::Real) = xval(d, -norminvlogcdf(lp))
 # quantile
-function quantile(d::Normal, p::Real)
+function quantile(d::Normal{T}, p::Real) where {T}
     if iszero(d.σ)
         if iszero(p)
-            -Inf
+            T(-Inf)
         elseif isone(p)
-            Inf
+            T(Inf)
         else
-            0.5
+            T(0.5)
         end
     end
-    xval(d, -erfcinv(2p) * sqrt2)
+    T(xval(d, -erfcinv(2p) * sqrt2))
```

Comment:
```suggestion
    xval(d, -erfcinv(2p) * sqrt2)
```
`xval` should do the right thing